### PR TITLE
feat: top_k.c translation

### PR DESCRIPTION
**Pull request by lhxsm**
If this PR is related to coding or code translation, please fill out the checklist and paste the console outputs to the PR.

最大的 3 个元素为: 堆的数组表示：[3, 6, 7]
堆的树状表示：
            /———7
———3
            \———6
- [x] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [x] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

### CODE REVIEW

File: codes/c/chapter_heap/top_k.c [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        result[i] = heap[i];
+    }
+
+    // 释放堆内存
+    free(heap);
+
+    return result;
+}
+
+int main() {
```
Comment (krahets) [inline codes/c/chapter_heap/top_k.c:44]: /* Driver Code */

File: codes/c/chapter_heap/top_k.c [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+ */
+
+#include "../utils/common.h"
+
+// 用于最小堆的比较函数
+int compare(const void *a, const void *b) {
+    return (*(int *)a - *(int *)b);
+}
+
+// 基于堆查找数组中最大的 k 个元素
```
Comment (krahets) [inline codes/c/chapter_heap/top_k.c:14]: /* */

**Review by krahets — COMMENTED** [review]
Hi @lhxsm , please check the code and comments format carefully to match other C code's style.

File: codes/c/chapter_heap/top_k.c [hunk 3]
```diff
@@ -41,6 +41,7 @@ int *topKHeap(int *nums, int size, int k) {
     return result;
 }
 
+// Driver Code
```
Comment (krahets) [inline codes/c/chapter_heap/top_k.c:44]: /* Driver Code */

Comment (krahets) [reply codes/c/chapter_heap/top_k.c:44]: /* Driver Code */ not // Driver Code

File: codes/c/chapter_heap/top_k.c [hunk 4]
```diff
@@ -0,0 +1,58 @@
+/**
+ * File: top_k.cpp
```
Comment (krahets) [inline codes/c/chapter_heap/top_k.c:2]: .c

**Comment by krahets** [comment]
Hi @lhxsm, please address the comments, thx!

**Comment by lhxsm** [comment]
> Hi @lhxsm, please address the comments, thx!

OK, I'm modifying it to use a reasonable heap structure to implement it

File: codes/c/chapter_heap/top_k.c [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+/* 基于堆查找数组中最大的 k 个元素 */
+int *topKHeap(int *nums, int size, int k) {
+    // 创建一个大小为 k 的最小堆    
+    int *heap = (int *)malloc(k * sizeof(int));
+    for (int i = 0; i < k; i++) {
+        heap[i] = nums[i];
+    }
+    qsort(heap, k, sizeof(int), compare);
```
[the remark this answers is not in the document]
Comment (krahets) [reply codes/c/chapter_heap/top_k.c:21]: Recommend collasing it after addressing the comment.

File: codes/c/chapter_heap/top_k.c [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    MinHeap *heap = createHeap();
+    for (int i = 0; i < k; i++) {
+        pushHeap(heap, nums[i]);
+    }
+
+    // 从第 k+1 个元素开始，保持堆的长度为 k
+    for (int i = k; i < size; i++) {
+        // 若当前元素大于堆顶元素，则将堆顶元素替换为当前元素，并重新调整堆
+        if (nums[i] > heap->data[0]) {
+            heap->data[0] = nums[i];
```
Comment (krahets) [inline codes/c/chapter_heap/top_k.c:54]: Heapifying is needed after replacing the element to maintain heap's property.

I feel like you don't know enough about the heap.

Comment (krahets) [reply codes/c/chapter_heap/top_k.c:54]: Cause we already implement a max heap in `max_heap.c`, you can solve this problem by this max heap and reverted integers.

**Please REFER TO the javascript implementation in `top_k.js`.**

**Comment by krahets** [comment]
See #889

**Pull request closed without merging**